addons/website/models/website.py

```diff
@@ -1340,3 +1341,235 @@ def _get_cached_values(self):
+        :param order: order on which to sort results within a model type
+
+        :return tuple containing:
+            - total number of results across all involved models
+            - list of results per model made of:
+                - initial search_detail for the model
+                - count: number of results for the model
+                - results: model list equivalent to a `model.search()`
+        """
+        all_results = []
+        total_count = 0
+        for search_detail in search_details:
+            model = self.env[search_detail['model']]
+            results, count = model._search_fetch(search_detail, search, limit, order)
+            search_detail['results'] = results
+            total_count += count
+            search_detail['count'] = count
+            all_results.append(search_detail)
+        return total_count, all_results
+
+    def _search_render_results(self, search_details, limit):
+        """
+        Prepares data for the autocomplete and hybrid list rendering
+
+        :param search_details: obtained from `_search_exact()`
+        :param limit: maximum number or rows to render
+
+        :return the updated `search_details` containing an additional `results_data` field equivalent
+            to the result of a `model.read()`
+        """
+        for search_detail in search_details:
+            fields = search_detail['fetch_fields']
+            results = search_detail['results']
+            icon = search_detail['icon']
+            mapping = search_detail['mapping']
+            results_data = results._search_render_results(fields, mapping, icon, limit)
+            search_detail['results_data'] = results_data
+        return search_details
+
+    def _search_find_fuzzy_term(self, search_details, search, limit=1000, word_list=None):
+        """
+        Returns the "closest" match of the search parameter within available words.
+
+        :param search_details: obtained from `_search_get_details()`
+        :param search: search term to which words must be matched against
+        :param limit: maximum number of records fetched per model to build the word list
+        :param word_list: if specified, this list of words is used as possible targets instead of
+            the words contained in the match fields of each involved model
+
+        :return: term on which a search can be performed instead of the initial search
+        """
+        def trigram_enumerate_words():
+            """
+            Browses through all words that need to be compared to the search term.
+            It extracts all words of every field associated to models in the fields_per_model parameter.
+            The search is restricted to a records having the non-zero pg_trgm.word_similarity() score.
+            """
+            match_pattern = '\\w{%s,}' % min(4, len(search) - 3)
+            search_term = "'%s'" % escape_psql(search)
```

Comment:
Why is the search term being quoted here?

Reply:
To be used as a SQL text literal (as opposed to a column name) when given to the unaccent and word_similarity functions

Reply:
Don't see why it'd need to be *quoted*, if it's correctly passed to `execute` (as a value) then it should automatically be handled correctly.

Reply:
It was because it was included as a literal within the "word_similarity(...)" expressions instead.
Now obsolete, I replaced everything with psycopg2.sql

Reply:
Ah ok.